### PR TITLE
ppl: fixes for annealing

### PR DESCRIPTION
**Pull request by eder-matheus**
Updates and fixes following the comments on PR https://github.com/The-OpenROAD-Project/OpenROAD/pull/3576.

### CODE REVIEW

File: src/ppl/src/SimulatedAnnealing.cpp [hunk 1]
```diff
@@ -185,9 +187,11 @@ int SimulatedAnnealing::randomAssignmentForGroups(
       pin_assignment_[pin_idx] = group_slot;
       slots_[group_slot].used = true;
       group_slot++;
-      slot_idx++;
       placed_pins.insert(pin_idx);
     }
+    while (slots_[slot_idx].used) {
+      slot_idx++;
+    }
```
Comment (maliberty) [inline src/ppl/src/SimulatedAnnealing.cpp:194]: This guarantees that you start on an empty slot but I'm not clear it prevents overlap (you might start the next group too close to the next group to fit).

**Review by maliberty — APPROVED** [review]

**Comment by maliberty** [comment]
conflicts to resolve

**Comment by eder-matheus** [comment]
> conflicts to resolve

Working on it. Also, I still see the different results in Mac vs Linux when using the `uniform_int_distribution`. Using the `uniform_real_distribution` is awkward in some cases, but at least we have consistent results. Should I keep the `uniform_real_distribution`, or accept the diff results when using `uniform_int_distribution`?

**Comment by maliberty** [comment]
> > conflicts to resolve
> 
> Working on it. Also, I still see the different results in Mac vs Linux when using the `uniform_int_distribution`. Using the `uniform_real_distribution` is awkward in some cases, but at least we have consistent results. Should I keep the `uniform_real_distribution`, or accept the diff results when using `uniform_int_distribution`?

Please look into why you get differences.  There should be no platform dependence with uniform_int_distribution afaik.  It just relies on the underlying generator for the randomness.

**Review by maliberty — APPROVED** [review]

**Pull request merged**